video/out/vo_vaapi_wayland.c

```diff
@@ -286,30 +304,39 @@ static int reconfig(struct vo *vo, struct mp_image_params *params) {
         int height = 1;
         int stride = MP_ALIGN_UP(width * 4, 16);
         int fd = allocate_memfd(stride);
-        if (fd < 0)
+        if (fd < 0) {
+            MP_ERR(vo, "Unable to allocate wl_shm pool file descriptor\n");
```

Comment:
consider making `allocate_memfd` itself print an useful error (including the errno)

Reply:
also it's duplicated between two files, maybe `wayland_common.c` is a good place for it:
<pre>
video/out/vo_wlshm.c
77:static int allocate_memfd(size_t size)
104:    fd = allocate_memfd(size);

video/out/vo_vaapi_wayland.c
230:static int allocate_memfd(size_t size) {
288:        int fd = allocate_memfd(stride);
</pre>

Reply:
The `memfd` check is a separate from the generic wayland checks. It's linux-only AFAIK, but I think freebsd copies it.

Reply:
I've moved the method to `wayland_common`